aikau/src/main/resources/alfresco/core/BaseWidget.js

```diff
@@ -0,0 +1,121 @@
+ * @since 1.0.100
+ */
+define(["dojo/_base/declare",
+        "dijit/_WidgetBase", 
+        "dijit/_TemplatedMixin",
+        "dojo/text!./templates/Label.html",
+        "alfresco/core/Core"], 
+        function(declare, _WidgetBase, _TemplatedMixin, template, Core) {
+   
+   return declare([_WidgetBase, _TemplatedMixin, Core], {
+
+      /**
+       * The template that defines the DOM for the widget. If configured this will 
+       * result in the capabilities of dijit/_TemplatedMixin being used to construct the 
+       * DOM. Ideally this should be left as the default value of null and the 
+       * [createWidgetDom]{@link module:alfresco/core/BaseWidget#createWidgetDom} function
+       * will be called - this function should be implemented to use the native capabilities
+       * of the browser to build a DOM for the widget.
+       * 
+       * @instance
+       * @type {string}
+       * @default
+       */
+      templateString: null,
+
+      /**
+       * This extends the function provided by the dijit/_WidgetBase module to prevent any 
+       * attributes from being applied. This changes the default widget behaviour in order
+       * to gain performance improvements. Any parameters that do need to be applied can
+       * be provided by overriding 
+       * [getParametersToApply]{@link module:alfresco/core/BaseWidget#getParametersToApply}.
+       * 
+       * @instance
+       */
+      _applyAttributes: function alfresco_core_BaseWidget___applyAttributes() {
+         var originalParams = this.params;
+         this.params = this.getParametersToApply();
+         this.inherited(arguments);
+         this.params = originalParams;
+      },
+
+      /**
+       * This is called from [_applyAttributes]{@link module:alfresco/core/BaseWidget#_applyAttributes}
+       * to get an object of the parameters that should be applied to the widget. This should only
+       * be implemented as necessary. The more parameters applied the worse the performance at scale.
+       * 
+       * @instance
+       * @overridable
+       * @param {object} originalParams The original parameters provided to the widget.
+       * @return {object} An object of key/value pair parameters to be applied. 
+       */
+      getParametersToApply: function alfresco_core_BaseWidget__getParametersToApply(/*jshint unused:false*/ originalParams) {
+         return null;
+      },
+
+      /**
+       * Extends the function inherited from dijit/_WidgetBase to only make use of the template
+       * rendering capabilities provided by dijit/_TemplatedMixin when a
+       * [templateString]{@link module:alfresco/core/BaseWidget#templateString} is provided. Otherwise
+       * the [createWidgetDom]{@link module:alfresco/core/BaseWidget#createWidgetDom} will be called
+       * to build the DOM for the widget.
+       * 
+       * @instance
+       */
+      buildRendering: function alfresco_core_BaseWidget__buildRendering() {
+         if (this.templateString)
```

Comment:
Before merging I'd like to refresh / check how _AttachMixin is tied into buildRendering cycle. I remember that for extensive DOM constructs (templated or otherwise) the lookup of any widgets / events to attach could get quite expensive and especially wastefull if there aren't any events / nodes to be attached at all or you are already handling it explicitly. I feel if we add the option for manual buildRendering we might as well include the same for attach handling - even if _TemplateMixin is used...